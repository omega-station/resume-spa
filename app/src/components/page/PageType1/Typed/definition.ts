export interface Props {
  strings: string[];
  cursorChar?: string;
  typeSpeed?: number;
}
