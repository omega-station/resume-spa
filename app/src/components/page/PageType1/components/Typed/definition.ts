export interface Props {
  /** Strings array */
  strings: string[];
  /** Cursor character */
  cursorChar?: string;
  /** Typing speed */
  typeSpeed?: number;
}
