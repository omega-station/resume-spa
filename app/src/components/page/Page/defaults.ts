import { Props } from './definition';

const defaults: Props = {
  type: 1,
};

export default defaults;
