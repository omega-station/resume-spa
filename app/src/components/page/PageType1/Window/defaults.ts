import { Props } from './definition';

const defaults: Props = {
  children: '',
  heading: 'Window Heading...',
};

export default defaults;
