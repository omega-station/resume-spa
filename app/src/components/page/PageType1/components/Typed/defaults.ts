import { Props } from './definition';

const defaults: Props = {
  strings: [
    `I will obey your orders. I will serve this ship as First Officer. And in an attack against the Enterprise, I will die with this crew. But I will not break my oath of loyalty to Starfleet. Sure. You'd be surprised how far a hug goes with Geordi, or Worf. Did you come here for something in particular or just general Riker-bashing? Maybe if we felt any human loss as keenly as we feel one of those close to us, human history would be far less bloody. Commander William Riker of the Starship Enterprise. I'll be sure to note that in my log. Shields up! Rrrrred alert! I guess it's better to be lucky than good. Talk about going nowhere fast. In all trust, there is the possibility for betrayal. Could someone survive inside a transporter buffer for 75 years? You enjoyed that. About four years. I got tired of hearing how young I looked. Why don't we just give everybody a promotion and call it a night - 'Commander'? The unexpected is our normal routine. We know you're dealing in stolen ore. But I wanna talk about the assassination attempt on Lieutenant Worf. The look in your eyes, I recognize it. You used to have it for me. This should be interesting. Some days you get the bear, and some days the bear gets you. Maybe if we felt any human loss as keenly as we feel one of those close to us, human history would be far less bloody. I recommend you don't fire until you're within 40,000 kilometers. Mr. Worf, you do remember how to fire phasers?`,
  ],
  cursorChar: 'â–“',
  typeSpeed: 0,
};

export default defaults;
