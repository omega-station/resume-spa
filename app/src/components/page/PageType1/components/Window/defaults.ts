import { Props } from './definition';

const defaults: Props = {
  children: '',
  heading: 'Window Heading...',
  isTest: false,
  onClickClose: () => {},
};

export default defaults;
