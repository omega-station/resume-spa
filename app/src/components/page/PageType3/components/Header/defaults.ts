import { Props } from './definition';

const defaults: Props = {
  section: 0,
  isMenuOpen: false,
  onClickMenuItem: () => {},
  onClickMenuToggle: () => {},
};

export default defaults;
