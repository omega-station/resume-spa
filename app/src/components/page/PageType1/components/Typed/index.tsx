import React from 'react';
import ReactTyped from 'react-typed';
import defaults from './defaults';
import { Props } from './definition';

const Typed = (props: Props): JSX.Element => {
  const { cursorChar, typeSpeed, strings }: Props = { ...defaults, ...props };
  return <ReactTyped cursorChar={cursorChar} strings={strings} typeSpeed={typeSpeed} />;
};

export default Typed;
