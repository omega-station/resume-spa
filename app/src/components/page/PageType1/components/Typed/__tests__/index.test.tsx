import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Typed from '..';
import defaults from '../defaults';

afterEach(cleanup);

describe('Typed', () => {
  it('should render with required strings={default} prop', () => {
    const component = render(<Typed strings={defaults.strings} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with required strings={default} and cursorChar={default} props', () => {
    const component = render(<Typed strings={defaults.strings} cursorChar={defaults.cursorChar} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with required strings={default} and typeSpeed={default} props', () => {
    const component = render(<Typed strings={defaults.strings} typeSpeed={defaults.typeSpeed} />);
    expect(component).toMatchSnapshot();
  });
});
