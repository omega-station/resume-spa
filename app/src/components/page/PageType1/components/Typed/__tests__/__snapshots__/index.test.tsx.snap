// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typed should render with required strings={default} and cursorChar={default} props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style
      data-typed-js-css="true"
      type="text/css"
    >
      
        .typed-cursor{
          opacity: 1;
        }
        .typed-cursor.typed-cursor--blink{
          animation: typedjsBlink 0.7s infinite;
          -webkit-animation: typedjsBlink 0.7s infinite;
                  animation: typedjsBlink 0.7s infinite;
        }
        @keyframes typedjsBlink{
          50% { opacity: 0.0; }
        }
        @-webkit-keyframes typedjsBlink{
          0% { opacity: 1; }
          50% { opacity: 0.0; }
          100% { opacity: 1; }
        }
      
    </style>
    <div>
      <span>
        <span />
        <span
          class="typed-cursor"
        >
          ▓
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <span />
      <span
        class="typed-cursor"
      >
        ▓
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Typed should render with required strings={default} and typeSpeed={default} props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style
      data-typed-js-css="true"
      type="text/css"
    >
      
        .typed-cursor{
          opacity: 1;
        }
        .typed-cursor.typed-cursor--blink{
          animation: typedjsBlink 0.7s infinite;
          -webkit-animation: typedjsBlink 0.7s infinite;
                  animation: typedjsBlink 0.7s infinite;
        }
        @keyframes typedjsBlink{
          50% { opacity: 0.0; }
        }
        @-webkit-keyframes typedjsBlink{
          0% { opacity: 1; }
          50% { opacity: 0.0; }
          100% { opacity: 1; }
        }
      
    </style>
    <div>
      <span>
        <span />
        <span
          class="typed-cursor"
        >
          ▓
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <span />
      <span
        class="typed-cursor"
      >
        ▓
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Typed should render with required strings={default} prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <span />
        <span
          class="typed-cursor"
        >
          ▓
        </span>
      </span>
    </div>
    <style
      data-typed-js-css="true"
      type="text/css"
    >
      
        .typed-cursor{
          opacity: 1;
        }
        .typed-cursor.typed-cursor--blink{
          animation: typedjsBlink 0.7s infinite;
          -webkit-animation: typedjsBlink 0.7s infinite;
                  animation: typedjsBlink 0.7s infinite;
        }
        @keyframes typedjsBlink{
          50% { opacity: 0.0; }
        }
        @-webkit-keyframes typedjsBlink{
          0% { opacity: 1; }
          50% { opacity: 0.0; }
          100% { opacity: 1; }
        }
      
    </style>
  </body>,
  "container": <div>
    <span>
      <span />
      <span
        class="typed-cursor"
      >
        ▓
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
